harness/determined/core/_train.py

```diff
@@ -268,3 +269,11 @@ def report_early_exit(self, reason: EarlyExitReason) -> None:
 
     def get_experiment_best_validation(self) -> Optional[float]:
         return None
+
+    def get_tensorboard_path(self) -> pathlib.Path:
+        # make an ephemeral directory for tensorboard tests
```

Comment:
This `DummyTrainContext()` will also show up for users in local training mode, so putting test-specific paths on this object seems dangerous.  Is there a way to accomplish this purely in tests without overriding this method in the public determined package?